src/o365/spo/commands/listitem/listitem-record-undeclare.ts

```diff
@@ -0,0 +1,305 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import {
+  CommandOption,
+  CommandValidate,
+  CommandTypes
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { Auth } from '../../../../Auth';
+import { ContextInfo, ClientSvcResponseContents, ClientSvcResponse } from '../../spo';
+
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+  id: string;
+
+}
+
+class SpoListItemRecordUndeclareCommand extends SpoCommand {
+  private objectIdentity: string;
+  constructor() {
+    super()/* istanbul ignore next */;
+    this.objectIdentity = '';
+  }
+  public get name(): string {
+    return commands.LISTITEM_RECORD_UNDECLARE;
+  }
+
+  public get description(): string {
+    return 'Undeclares  listitem  as a record';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = typeof args.options.listId !== 'undefined';
+    telemetryProps.listTitle = typeof args.options.listTitle !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    const listIdArgument = args.options.listId || '';
+    const listTitleArgument = args.options.listTitle || '';
+    let siteAccessToken: string = '';
+    const listRestUrl: string = (args.options.listId ?
+      `${args.options.webUrl}/_api/web/lists(guid'${encodeURIComponent(listIdArgument)}')`
+      : `${args.options.webUrl}/_api/web/lists/getByTitle('${encodeURIComponent(listTitleArgument)}')`);
+
+
+    let formDigestValue: string = '';
+    let environmentListId: string = '';
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise | Promise<any> => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}.`);
+          cmd.log(``);
+          cmd.log(`auth object:`);
+          cmd.log(auth);
+        }
+        if (this.verbose) {
+          cmd.log(`Getting list id...`);
+        }
+
+          const listRequestOptions: any = {
```

Comment:
If we already have the list ID, there is no need to execute this request. Instead, we could return a resolved promise with the list ID passed in the `listId` command option.